Resources/Core/Functional/Extensions/json_response/Classes/Middleware/BackendUserHandler.php

```diff
@@ -55,6 +55,10 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
             // Init backend user if found in database
             $backendUser = GeneralUtility::makeInstance(FrontendBackendUserAuthentication::class);
             $backendUser->user = $row;
+            $backendUser->uc = isset($row['uc']) ? unserialize($row['uc']) : [];
+            $backendUser->initializeUserSessionManager();
+            $backendUser->fetchGroupData();
+            $workspaceId = $workspaceId ?? 0;
```

Comment:
This literlly renders the condition on the next line use-less. Beside that, I'd say that we should go here more for using `$backendUser->setDefaultWorkspace()`, not fallback to `0` and only set the temporary workspace id if specified in the internal request context.

What do you think ?

Reply:
Ah sure, didn't notice that one.

Only setting if provided would not solve the issue. I would need to dig into it once more. But the issue arised by not providing a workspace id.

Reply:
never mind - `setDefaultWorkspace()` is protected. However, the if condition should be removed and setTemporaryWorkspace() always called with the 0 fallback.

```php
        if ($row !== false) {
            // Init backend user if found in database
            $backendUser = GeneralUtility::makeInstance(FrontendBackendUserAuthentication::class);
            $backendUser->user = $row;
            $backendUser->uc = isset($row['uc']) ? unserialize($row['uc']) : [];
            $backendUser->initializeUserSessionManager();
            $backendUser->fetchGroupData();
            $backendUser->setTemporaryWorkspace($workspaceId ?? 0);
            $GLOBALS['BE_USER'] = $backendUser;
            $this->setBackendUserAspect(GeneralUtility::makeInstance(Context::class), $backendUser);
        }
```

Reply:
Sure, done.